src/snagrecover/protocols/rockchip.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2025 Collabora Ltd.
+#
+# SPDX-License-Identifier: GPL-2.0+
+#
+# Author: Arnaud Patard <arnaud.patard@collabora.com>
+
+
+from crccheck.crc import Crc16Ibm3740
+import logging
+import usb
+
+from snagrecover import utils
+
+logger = logging.getLogger("snagrecover")
+
+class RochipBootRomError(Exception):
+	pass
+
+class RochipBootRom():
+	def __init__(self, dev: usb.core.Device):
+		self.dev = dev
+		cfg = dev.get_active_configuration()
+
+		eps_found = False
+		for intf in cfg.interfaces():
+			ep_in, ep_out = None, None
+			for ep in intf.endpoints():
+				is_bulk = (ep.bmAttributes & usb.ENDPOINT_TYPE_MASK) == usb.ENDPOINT_TYPE_BULK
+				is_in = (ep.bmAttributes & usb.ENDPOINT_TYPE_MASK) == usb.ENDPOINT_TYPE_BULK
+				if not is_bulk:
+					continue
+				is_in = (ep.bEndpointAddress & usb.ENDPOINT_DIR_MASK) == usb.ENDPOINT_IN
+				if is_in:
+					ep_in = ep.bEndpointAddress
+				else:
+					ep_out = ep.bEndpointAddress
+			if not ((ep_in is None) or (ep_out is None)):
+				eps_found = True
+				break
+		if not eps_found:
+			raise RochipBootRomError("No BULK IN/OUT endpoint pair found in device")
+		self.ep_in = ep_in
+		self.ep_out = ep_out
```

Comment:
You do not really need the `ep_in`/`ep_out` for the `ctrl_transfer` done here, these endpoints are mostly needed for the RockUSB protocol, something that is not supported or used here.